src/status_im/events.cljs

```diff
@@ -441,18 +446,138 @@
+                           :on-accept           #(re-frame/dispatch [:chat.ui/clear-history])}}))
 
 (handlers/register-handler-fx
- :chat.ui/delete-chat-pressed
- (fn [_ [_ chat-id]]
-   {:ui/show-confirmation {:title (i18n/label :t/delete-chat-confirmation)
-                           :content ""
-                           :confirm-button-text (i18n/label :t/delete-chat-action)
-                           :on-accept #(re-frame/dispatch [:remove-chat-and-navigate-home chat-id])}}))
+ :chat.ui/remove-chat-pressed
+ (fn [_ [_ chat-id group?]]
+   {:ui/show-confirmation {:title               (i18n/label :t/delete-confirmation)
+                           :content             (i18n/label :t/delete-chat-confirmation)
+                           :confirm-button-text (i18n/label :t/delete)
+                           :on-accept           #(re-frame/dispatch [:chat.ui/remove-chat chat-id])}}))
+
+(handlers/register-handler-fx
+ :chat.ui/set-chat-ui-props
+ (fn [{:keys [db]} [_ kvs]]
+   {:db (chat/set-chat-ui-props db kvs)}))
+
+(handlers/register-handler-fx
+ :chat.ui/show-message-details
+ (fn [{:keys [db]} [_ details]]
+   {:db (chat/set-chat-ui-props db {:show-bottom-info? true
+                                    :bottom-info       details})}))
+
+(handlers/register-handler-fx
+ :chat.ui/show-message-options
+ (fn [{:keys [db]} [_ options]]
+   {:db (chat/set-chat-ui-props db {:show-message-options? true
+                                    :message-options       options})}))
+
+(handlers/register-handler-fx
+ :chat.ui/navigate-to-chat
+ (fn [cofx [_ chat-id opts]]
+   (chat/navigate-to-chat cofx chat-id opts)))
+
+(handlers/register-handler-fx
+ :chat.ui/load-more-messages
+ [(re-frame/inject-cofx :data-store/get-messages)
+  (re-frame/inject-cofx :data-store/get-user-statuses)]
+ (fn [cofx _]
+   (chat.loading/load-more-messages cofx)))
+
+(handlers/register-handler-fx
+ :chat.ui/start-chat
+ (fn [cofx [_ contact-id opts]]
+   (chat/start-chat cofx contact-id opts)))
+
+(handlers/register-handler-fx
+ :chat.ui/start-public-chat
+ (fn [cofx [_ topic modal?]]
+   (chat/start-public-chat cofx topic modal?)))
+
+(handlers/register-handler-fx
+ :chat.ui/start-group-chat
+ (fn [cofx [_ group-name]]
+   (chat/start-group-chat cofx group-name)))
+
+(handlers/register-handler-fx
+ :chat.ui/remove-chat
+ (fn [cofx [_ chat-id]]
+   (chat/remove-chat cofx chat-id)))
+
+(handlers/register-handler-fx
+ :chat.ui/clear-history
+ (fn [{{:keys [current-chat-id]} :db :as cofx} _]
+   (chat/clear-history cofx current-chat-id)))
+
+(handlers/register-handler-fx
+ :chat.ui/resend-message
+ (fn [cofx [_ chat-id message-id]]
+   (chat.message/resend-message cofx chat-id message-id)))
+
+(handlers/register-handler-fx
+ :chat.ui/delete-message
+ (fn [cofx [_ chat-id message-id]]
+   (chat.message/delete-message cofx chat-id message-id)))
+
+(handlers/register-handler-fx
+ :chat.ui/show-profile
+ (fn [cofx [_ identity]]
+   (navigation/navigate-to-cofx
```

Comment:
should be a navigate-to-profile function that does this in a fx/merge

Reply:
Good point, will change that.

Reply:
Will be done when the profile module is refactored into agreed pattern.